src/SegmentedArray.chpl

```diff
@@ -717,6 +717,78 @@ module SegmentedArray {
       return (subbedOffsets, subbedVals, numReplacements);
     }
 
+    /*
+      Strip out all of the leading and trailing characters of each element of a segstring that are
+      called out in the "pattern" argument.
+
+      :arg pattern: pattern used to find matches of the leading and trailing characters in the segstring
+      :type pattern: string
+
+      :arg containsSpace: the string to replace pattern matches with
+      :type replStr: string
+
+      :arg initCount: If count is nonzero, at most count splits occur. If zero, substitute all occurences of pattern
+      :type initCount: int
+
+      :arg containsSpace: If True, appends a space character to the pattern
+      :type containsSpace: bool
+
+      :returns: Strings – substrings with stripped characters from the original string and the offsets into those substrings
+    */
+
+    proc strip(pattern: string, containsSpace: bool) throws {
+      checkCompile(pattern);
+      ref origOffsets = this.offsets.a;
+      ref origVals = this.values.a;
+      const lengths = this.getLengths();
+
+      saLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+                                             "pattern: %s - origOffsets: %t - origVals: %t"
+                                             .format(pattern, origOffsets, origVals:bytes));
+
+      var replacedLens: [this.offsets.aD] int;
+      var newPattern: string;
+
+      // Pattern contains only whitespace characters or is empty
+      if containsSpace == true {
+        newPattern = pattern + " ";
+      }
+
+      saLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),"pattern: (%t)".format(newPattern));
+
+      forall (off, len, rlen) in zip(origOffsets, lengths, replacedLens) with (var valAgg = newDstAggregator(uint(8))) {
```

Comment:
`with (var valAgg = newDstAggregator(uint(8))` is not needed here because you do not use the `valAgg` until the next loop.

Reply:
Yep. Removed.